Add dam com1 dfa (#756)
avaframe/AvaFrame: docs/damToolCom1DFA.rst

Dam tool com1DFA
=================

The com1DFA computational module provides the option to take the effect of dams into account.

> This is done using a haddock method based on particles bouncing on a dam wall.

Reviewer: ```suggestion
This is done using an ad-hoc method based on particles bouncing on a dam wall.
```

Reviewer: maybe bouncing back or being reflected/deflected by a dam wall?

Reviewer: and maybe based on the idea of particles ---

Input
-------

The dam is described by the crown line, the slope and the restitution coefficient:

>   * crown line as shape file (use the line type and enable the enable the "additional dimentions" option in order

Reviewer: ```suggestion
  * crown line as shape file (use the line type and enable the "additional dimensions" option in order
```

>     to specify the z coordinate). The z coordinate corresponds to the absolute height. The dam is then located on the

Reviewer: ```suggestion
    to specify the z coordinate). The z coordinate corresponds to the absolute height (terrain elevation plus dam height). The dam is then located on the
```

>     left side of the dam. The dam shape files lives in the ``avaDir/Inputs/DAM/`` directory.

Reviewer: ```suggestion
    left side of the dam. The dam shape files live in the ``avaDir/Inputs/DAM/`` directory. 
```

Reviewer: only one allowed? maybe add this info

    reflection) to be specified in the ini file (default value is 0)

Theory
--------

Dam description
~~~~~~~~~~~~~~~~

The dam is described by a crown line, that is to say a series of x, y, z points describing the crown of
the dam (the dam wall is located on the left side of the line), by the slope of the dam wall
(slope measured from the horizontal, :math:`\beta`) and a restitution coefficient (describing if we consider
more elastic or inelastic collisions between the particles and the dam wall, varying between 0 and 1).

The geometrical description of the dam is given on the figure :numref:`fig-DamToolSide`.
The dam crown line (:math:`\mathbf{x_\text{crown}}`) is projected onto the topography, which provides us
with the dam center line (:math:`\mathbf{x_\text{center}}`). We compute the tangent vector to the
center line (:math:`\mathbf{t_f}`). From this tangent vector and the dam slope, it is possible to
compute the wall tangent vector (:math:`\mathbf{t_w}`). Knowing the wall tangent vector and height,
it is possible to determine normal vector to the wall (:math:`\mathbf{n_w}`) and the foot line which

> is intersection between the dam wall and the topography (:math:`\mathbf{x_\text{foot}}`).

Reviewer: ```suggestion
is the intersection between the dam wall and the topography (:math:`\mathbf{x_\text{foot}}`).
```

When the dam fills up (flow thickness increases), the foot line is modified
(:math:`\mathbf{x_\text{foot}^\text{filled}} = \mathbf{x_\text{foot}} + \frac{h_v}{2} \mathbf{e_z}`).

> The normal and tangent vectors to the dam wall is readjusted accordingly.

Reviewer: ```suggestion
The normal and tangent vectors to the dam wall are readjusted accordingly.
```

.. _fig-DamToolSide:

.. figure:: _static/damToolSideView.png
          :width: 90%

>           Side view of the dam (cut view). :math:`\mathbf{x_\text{crown}}` describe the crown

Reviewer: ```suggestion
          Side view of the dam (cut view). :math:`\mathbf{x_\text{crown}}` describes the crown
```

          of the dam, :math:`\mathbf{x_\text{center}}` is the vertical projection of the crown
          on the topography (here the light blue line represents the topography).
          The tangent vector to the center line (:math:`\mathbf{t_f}`) is computed from the
          center line points. The tangent vector to the center line with the dam slope angle enable to compute
          the tangent (:math:`\mathbf{t_w}`) and normal (:math:`\mathbf{n_w}`) vector to the dam wall.
          Finally, this normal vector is adjusted depending on the snow thickness at the dam location

>           (filling of the dam)

Reviewer: (filling of the dam, n_w _filled).
